megabeast/settings.py

```diff
@@ -0,0 +1,81 @@
+__all__ = ["settings"]
+
+
+class settings:
+    """
+    All of the settings for the MegaBEAST.
+
+    Attributes
+    ----------
+    settings_file : string
+        name of the parameter file that contains the settings in python format
+        (e.g., supports dictonaries)
+    """
+
+    def __init__(self, settings_file):
+        """
+        Parameters
+        ----------
+        settings_file : string
+            input file name
+        """
+
+        self.settings_file = settings_file
+
+        if self.settings_file is not None:
+            # read in the file
+            self.read_settings()
+            # verify parameters
+            self.verify_settings()
+
+    def read_settings(self):
+        """
+        Read in the settings file and set parameters
+        """
+
+        # read everything in as strings
+        with open(self.settings_file, "r") as f:
+            temp_data = f.readlines()
+        # remove empty lines and comments
+        input_data = [
+            line.strip()
+            for line in temp_data
+            if line.strip() != "" and line.strip()[0] != "#"
+        ]
+        # remove comments that are mid-line (e.g., "x = 5 #comment")
+        for i, line in enumerate(input_data):
+            try:
+                input_data[i] = line[: line.index("#")]
+            except ValueError:
+                pass
+        # if parameters are defined over multiple lines, combine lines
+        for i in reversed(range(len(input_data))):
+            if ("import " not in input_data[i]) and ("=" not in input_data[i]):
+                input_data[i - 1] += input_data[i]
+                del input_data[i]
+
+        # parse it into a dictionary
+        beast_params = {}
+
+        for i in range(len(input_data)):
```

Comment:
Codacy found an issue: [Consider using enumerate instead of iterating with range and len](https://app.codacy.com/manual/kgordon/megabeast/pullRequest?prid=6302069)